examples/ffa_simulation.rs

```diff
@@ -0,0 +1,224 @@
+
+    for (id, (pos, s)) in &mut world.query::<(&mut Position, &Speed)>() {
+        let change = (
+            rng.gen_range(-s.speed, s.speed),
+            rng.gen_range(-s.speed, s.speed),
+        );
+        pos.x += change.0;
+        pos.y += change.1;
+        println!("Unit {:?} moved to {:?}", id, pos);
+    }
+}
+
+// In this system entities find the closest entity and fire at them
+fn system_fire_at_closest(world: &mut World) {
+    for (id0, (pos0, dmg0, kc0)) in
+        &mut world.query::<With<Health, (&Position, &Damage, &mut KillCount)>>()
+    {
+        let mut min_dist: Option<i32> = None;
+        let mut closest_id: Option<Entity> = None;
+
+        // find closest:
+        for (id1, pos1) in &mut world.query::<With<Health, &Position>>() {
+            if id0 != id1 {
+                let dist = manhattan_dist(pos0.x, pos1.x, pos0.y, pos1.y);
+                match min_dist {
+                    None => {
+                        min_dist = Some(dist);
+                    }
+                    Some(mut _dist0) => {
+                        _dist0 = _dist0.min(dist);
+                    }
+                }
+                if Some(dist) == min_dist {
+                    closest_id = Some(id1);
+                }
+            }
+        }
+
+        if !closest_id.is_some() {
+            println!("{:?} is the last survivor!", id0);
+            return;
+        }
+
+        // Deal damage:
+        /*
+                // Get target unit hp like this:
+                let mut hp1 = world.query_one::<&mut Health>(closest_id.unwrap()).unwrap();
+                let hp1 = hp1.get().unwrap();
+        */
+        // Or like this:
+        let mut hp1 = world.get_mut::<Health>(closest_id.unwrap()).unwrap();
+
+        // Is target unit still alive?
+        if hp1.hp > 0 {
+            // apply damage
+            hp1.hp = hp1.hp - dmg0.dmg;
+            println!(
+                "Unit {:?} was damaged by {:?} for {:?} HP",
+                closest_id.unwrap(),
+                id0,
+                dmg0.dmg
+            );
+            if hp1.hp <= 0 {
+                // if this killed it, increase own killcount
+                kc0.count += 1;
+                println!(
+                    "Unit {:?} was killed by unit {:?}!",
+                    closest_id.unwrap(),
+                    id0
+                );
+            }
+        }
+    }
+}
+
+fn system_remove_dead(world: &mut World) {
+    // Here we query entities with 0 or less hp and despawn them
+    let mut to_remove: Vec<Entity> = Vec::new();
+    for (id, hp) in &mut world.query::<&Health>() {
+        if hp.hp <= 0 {
+            to_remove.push(id);
+        }
+    }
+
+    for i in 0..to_remove.len() {
+        world.despawn(to_remove[i]).unwrap();
+    }
+}
+
+fn print_world_state(world: &mut World) {
+    for (id, (hp, pos, dmg, kc)) in &mut world.query::<(&Health, &Position, &Damage, &KillCount)>()
+    {
+        println!("Entity stats:");
```

Comment:
moving this print statement outside of the loop would make the output much cleaner IMHO

```
Entity stats:
ID: 4v0, Health { hp: 39 }, Damage { dmg: 9 }, Position { x: -15, y: 0 }, KillCount { count: 0 }
ID: 1v0, Health { hp: 41 }, Damage { dmg: 8 }, Position { x: 2, y: 3 }, KillCount { count: 0 }
ID: 2v0, Health { hp: 39 }, Damage { dmg: 2 }, Position { x: -14, y: -5 }, KillCount { count: 0 }
ID: 3v0, Health { hp: 30 }, Damage { dmg: 7 }, Position { x: 0, y: 7 }, KillCount { count: 1 }
```